src/components/Navigation.js

```diff
@@ -169,22 +213,196 @@ function Navigation() {
+                placeholder={`Search here`}
                 aria-label="Search"
                 ref={txSearchRef}
                 onKeyUp={handleKeyUp}
               />
-              <i className="fa fa-search pointer" onClick={search}></i>
             </div>
-            <Version />
           </div>
         </div>
+        <div className="mainnet-container hide-tabs">
+          <img
+            src={theme === 'dark' ? sun : moon}
+            alt="themeColorButton"
+            className="theme-color-btn"
+            onClick={() => dispatch(toggleTheme())}
+          />
+          <button className="mainnet-btn">
+            <GlobeMainnet
+              className={`${
+                theme === 'dark' ? 'dark-theme-logo' : 'light-theme-logo'
+              } theme-mainnet-btn`}
+            />
+            <span className="nav-title">Hathor Mainnet</span>
+          </button>
+        </div>
+        <div className="mobile-tabs">
+          {showSearchInput ? (
+            <input
+              type="search"
+              className={`form-control me-2 bg-dark text-light mobile-search-input ${
+                showSearchInput ? 'expanded' : ''
+              }`}
+              placeholder="Search..."
+              aria-label="Search"
+              ref={txSearchRef}
+              onKeyUp={handleKeyUp}
+              onBlur={() => setShowSearchInput(false)}
+              autoFocus
+            />
+          ) : (
+            <SearchIcon
+              fill={theme === 'dark' ? 'white' : 'black'}
+              onClick={toggleSearchInput}
+              className="mobile-search-icon"
+            />
+          )}
+          <MenuIcon
+            fill={theme === 'dark' ? 'white' : 'black'}
+            onClick={showSidebarHandler}
+            className="mobile-sidebar-icon"
+          />
+        </div>
+        <Sidebar close={() => setShowSidebar(false)} open={showSidebar} />
       </nav>
-      <HathorAlert ref={alertErrorRef} text="Invalid hash format or address" type="danger" />
-    </div>
-  );
+    );
+  };
+
+  const renderUi = () => {
+    return (
+      <div className="main-nav">
+        <nav className="navbar navbar-expand-lg navbar-dark ps-3 ps-lg-0">
+          <div className="d-flex flex-column align-items-center">
+            <Link className="navbar-brand" to="/" href="/">
+              <img src={logo} alt="" />
+            </Link>
+          </div>
+          <button
+            className="navbar-toggler"
+            type="button"
+            data-bs-toggle="collapse"
+            data-bs-target="#navbarSupportedContent"
+            aria-controls="navbarSupportedContent"
+            aria-expanded="false"
+            aria-label="Toggle navigation"
+          >
+            <span className="navbar-toggler-icon" />
+          </button>
+          <div className="collapse navbar-collapse" id="navbarSupportedContent">
+            <ul className="navbar-nav me-auto">
```

Comment:
Correct me if i'm wrong, but it seems there is a lot of duplicated items in the `renderUi`, `renderNewUi` and `SideBar`.
Maybe we should create a component for these duplicated items.

For instance, this list (`ul`) seems like it would have no problem being extracted to another component.
What do you think?

Reply:
I agree, but since we focused on customizing the existing UI for the proposed revamp, aiming to minimize changes (even in terms of componentization), could we consider postponing this refactor for now?